src/main/scala/firrtl/passes/Passes.scala

```diff
@@ -198,6 +198,15 @@ object Legalize extends Pass {
       e
     }
   }
+  private def legalizeShiftLeft(e: DoPrim): Expression = {
+    require(e.op == Shl)
+    val amount = e.consts.head.toInt
+    if (amount < 0) {
+      error(s"Argument $amount < 0 for Primop Shift Left")
+     } else {
+      e
+    }
+  }
```

Comment:
I think this check should go in CheckHighForm. Legalize (admittedly poorly named) is intended to include constant prop-like logic that is required for Verilog emission (as opposed to just an optimization).